compiler/il/OMRNode.cpp

```diff
@@ -257,6 +261,7 @@ OMR::Node::Node(TR::Node * from, uint16_t numChildren)
    self()->setReferenceCount(from->getReferenceCount());
    self()->setVisitCount(from->getVisitCount());
    self()->setLocalIndex(from->getLocalIndex());
+   self()->setKnownObjectIndex(from->getKnownObjectIndex());
```

Comment:
I guess we use this constructor when duplicating. What are the node flags of the new node? If they're clear, then `_knownObjectIndex` should be `UNKNOWN`. Maybe they're not clear though if e.g. `duplicateTree()` comes through here (since otherwise `duplicateTreeForCodeMotion()` wouldn't be needed)

Reply:
When you say "clear" you mean none of them are set?  What's an example of a node flag that must be set when there is a known object index?  If we set the known index on a node then does this imply some node flag must then be set to accompany it?

Reply:
There's no dependency between the known object index and any particular flag. The known object index is just a piece of information that has possibly been determined due to flow-sensitive analysis. If we're creating a `Node` where the flags are all zero, that means we don't know anything about the result (until/unless the caller attaches some information that it knows to be true)

Reply:
That's the same reason why I'm saying it should be reset in `resetFlagsForCodeMotionHelper()`

Reply:
OK.  Thanks for pointing that function out.  I'll leave this code here in the constructor but ensure the known object info does not get duplicated in `resetFlagsForCodeMotionHelper()`.

Reply:
Pushed an update.  I renamed `resetFlagsForCodeMotionHelper()` to reflect the fact that it can reset properties as well.